addons/procurement/procurement.py

```diff
@@ -60,177 +52,143 @@ class procurement_rule(osv.osv):
-
-        # These two fields are used for shceduling
-        'priority': fields.selection(PROCUREMENT_PRIORITIES, 'Priority', required=True, select=True, track_visibility='onchange'),
-        'date_planned': fields.datetime('Scheduled Date', required=True, select=True, track_visibility='onchange'),
-
-        'group_id': fields.many2one('procurement.group', 'Procurement Group'),
-        'rule_id': fields.many2one('procurement.rule', 'Rule', track_visibility='onchange', help="Chosen rule for the procurement resolution. Usually chosen by the system but can be manually set by the procurement manager to force an unusual behavior."),
-
-        'product_id': fields.many2one('product.product', 'Product', required=True, states={'confirmed': [('readonly', False)]}, readonly=True),
-        'product_qty': fields.float('Quantity', digits_compute=dp.get_precision('Product Unit of Measure'), required=True, states={'confirmed': [('readonly', False)]}, readonly=True),
-        'product_uom': fields.many2one('product.uom', 'Product Unit of Measure', required=True, states={'confirmed': [('readonly', False)]}, readonly=True),
-
-        'state': fields.selection([
-            ('cancel', 'Cancelled'),
-            ('confirmed', 'Confirmed'),
-            ('exception', 'Exception'),
-            ('running', 'Running'),
-            ('done', 'Done')
-        ], 'Status', required=True, track_visibility='onchange', copy=False),
-    }
-
-    _defaults = {
-        'state': 'confirmed',
-        'priority': '1',
-        'date_planned': lambda *a: time.strftime('%Y-%m-%d %H:%M:%S'),
-        'company_id': lambda self, cr, uid, c: self.pool.get('res.company')._company_default_get(cr, uid, 'procurement.order', context=c)
-    }
-
-    def _needaction_domain_get(self, cr, uid, context=None):
-        return [('state', '=', 'exception')] 
-        
-    def unlink(self, cr, uid, ids, context=None):
-        procurements = self.read(cr, uid, ids, ['state'], context=context)
-        unlink_ids = []
-        for s in procurements:
-            if s['state'] == 'cancel':
-                unlink_ids.append(s['id'])
-            else:
-                raise UserError(_('Cannot delete Procurement Order(s) which are in %s state.') % s['state'])
-        return osv.osv.unlink(self, cr, uid, unlink_ids, context=context)
-
-    def do_view_procurements(self, cr, uid, ids, context=None):
+
+    name = fields.Text(string='Description', required=True)
+    origin = fields.Char(string='Source Document', help="Reference of the document that created this Procurement. This is automatically completed by Odoo.")
+    company_id = fields.Many2one('res.company', string='Company', required=True, default=lambda self: self.env.user.company_id)
+    # These two fields are used for shceduling
+    priority = fields.Selection(PROCUREMENT_PRIORITIES, required=True, select=True, track_visibility='onchange', default='1')
```

Comment:
**select** should be renamed to **index** in the new API (to have a database index on the column).